### PR TITLE
Fix `numpy.unwrap` for NumPy 1.21

### PR DESCRIPTION
**Pull request by emcastillo**
For Numpy 1.21 compat

### CODE REVIEW

**Comment by kmaehashi** [comment]
In NumPy 1.21 the dtype is derived from `p` and new option `period`, so seems it can be even an integer.
https://github.com/numpy/numpy/pull/16987/files#diff-077cb9d49365526e8128efeeca9d2cf1646bab14b7c3125d0282603d35ea0fbeR1558

**Comment by emcastillo** [comment]
Thanks for the pointer! I should have checked the numpy documentation first ...
I synced the routine to that of numpy now, should be ok :).

I will add tests tomorrow

**Comment by emcastillo** [comment]
Tests added, wonder if we should parameterize them or it is fine to leave them as they are.

File: cupy/_math/trigonometric.py [hunk 1]
```diff
@@ -130,11 +135,26 @@ def unwrap(p, discont=numpy.pi, axis=-1):
...
     dd = sumprod.diff(p, axis=axis)
+    if discont is None:
+        discont = period/2
     slice1 = [slice(None, None)]*nd     # full slices
     slice1[axis] = slice(1, None)
     slice1 = tuple(slice1)
-    ph_correct = _unwrap_correct(dd, discont)
-    up = cupy.array(p, copy=True, dtype='d')
+    dtype = numpy.result_type(dd.dtype, period)
+    if numpy.core.numeric.issubdtype(dtype, numpy.core.numeric.integer):
```
Comment (kmaehashi) [inline cupy/_math/trigonometric.py:144]: Let's use documented API :smile:
```suggestion
    if numpy.issubdtype(dtype, numpy.core.numeric.integer):
```

**Review by kmaehashi — CHANGES_REQUESTED** [review]
Could you fix the docs indentation?
https://cupy--5385.org.readthedocs.build/en/5385/reference/generated/cupy.unwrap.html

**Comment by emcastillo** [comment]
should be ok now :)

**Comment by emcastillo** [comment]
Docs preview:

https://cupy--5385.org.readthedocs.build/en/5385/reference/generated/cupy.unwrap.html#cupy.unwrap

**Comment by kmaehashi** [comment]
LGTM!
pfnCI, test this please.

**Review by kmaehashi — APPROVED** [review]

**Pull request merged**